### ISSUE TITLE
[Feature] Better support for JSON data

### ISSUE THREAD

**Issue by JingsongLi** [apache/incubator-paimon#1390]
### Search before asking

- [X] I searched in the [issues](https://github.com/apache/incubator-paimon/issues) and found nothing similar.

### Motivation

We should support JSON inputs with schema evolution.

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR!

**Comment by reswqa** [comment]
Is it means to support `JsonFileFormat`? If not, could you provide a more detailed explanation of this ticket, thanks.

**Comment by JingsongLi** [comment]
> Is it means to support `JsonFileFormat`? If not, could you provide a more detailed explanation of this ticket, thanks.

No, I just mean how to write json records to Paimon, at the beginning, there is no schema definition, and as the json data changes, the columns should be automatically added to the paimon table.

**Comment by JingsongLi** [comment]
related to #1392

**Comment by MonsterChenzhuo** [comment]
@JingsongLi  please assign to me; thx。

**Comment by AnemoneIndicum** [comment]
Do you consider adding new data types for JSON formatted data?

**Comment by HeChuanXUPT** [comment]
我也需要一个json的输入，类似MongoDB
1. json数据输入字段是不定的，可增加
2. 读取的时候才指定字段
3. 读取的时候提供检索字段的能力


I need input format for JSON data , similar to MongoDB
1. JSON data fields are variable and can be added
2. Specify fields when reading
3. Provide the ability to retrieve fields during reading
